compiler/rustc_span/src/hygiene.rs

```diff
@@ -1223,6 +1240,7 @@ pub fn register_expn_id(
     data: ExpnData,
     hash: ExpnHash,
 ) -> ExpnId {
+    debug_assert!(data.parent == ExpnId::root() || krate == data.parent.krate);
```

Comment:
Should this be a regular assert?

Reply:
Seems fine as is, the assert will be removed anyway when root expansions start keeping their crate IDs.